Extensions/Lighting/lightruntimeobject.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * @typedef {Object} LightObjectDataType
+ * @property {Object} content The base parameters of light object.
+ * @property {number} content.radius The radius of light object.
+ * @property {string} content.color A string representing color in hexadecimal format.
+ * @property {string} content.texture A string representing the name of texture used for light object.
+ * @property {boolean} content.debugMode true if the light objects shows debug graphics, false otherwise.
+ *
+ * @typedef {ObjectData & LightObjectDataType} LightObjectData
+ */
+
+/**
+ * Displays a Light object.
+ * @memberof gdjs
+ * @class LightRuntimeObject
+ * @extends RuntimeObject
+ * @param {gdjs.RuntimeScene} runtimeScene
+ * @param {LightObjectData} lightObjectData
+ */
+gdjs.LightRuntimeObject = function (runtimeScene, lightObjectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, lightObjectData);
+
+  /** @type {number} */
+  this._radius =
+    lightObjectData.content.radius > 0 ? lightObjectData.content.radius : 1;
+
+  /** @type {number[]} color in format [r, g, b], where each component is in the range [0, 255] */
+  this._color = gdjs.LightRuntimeObject.hexToRGBColor(
+    lightObjectData.content.color
+  );
+
+  /** @type {boolean} */
+  this._debugMode = lightObjectData.content.debugMode;
+
+  /** @type {?PIXI.Texture} */
+  this._texture =
+    lightObjectData.content.texture === ''
+      ? null
+      : runtimeScene
+          .getGame()
+          .getImageManager()
+          .getPIXITexture(lightObjectData.content.texture);
+
+  /** @type {?gdjs.LightObstaclesManager} */
+  this._obstaclesManager =
+    gdjs.LightObstaclesManager !== undefined
+      ? gdjs.LightObstaclesManager.getManager(runtimeScene)
```

Comment:
I believe there is no reason for this to be undefined anymore? We should move this to be a ` @type {gdjs.LightObstaclesManager}` then.